go/fn/run.go

```diff
@@ -27,7 +27,14 @@ import (
 // AsMain reads the resourceList in yaml format from stdin, evaluates the
 // function and write the updated resourceList in yaml to stdout. Errors if any
 // will be printed to stderr.
-func AsMain(p ResourceListProcessor) error {
+func AsMain(input interface{}) error {
```

Comment:
Please add godoc for this. It's not obvious what the input should be when looking at the function signature.

Reply:
Done. Thanks